### PR TITLE
Pass full sources response to labs

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Previously, we were passing the just the `source` portion of the main.json file to labs. However, in order to enable artist variants in music lab, we'll need the full contents of the response to access other properties. Additionally as we support other labs, we'll probably want access to other properties (like animations, html, etc). This modifies lab lab code to pass down the full object rather than just the source portion.

I also had a couple changes on another branch which felt relevant to pull in here - used HttpClient to preserve types when making API requests, and streamlining some of the loading logic a bit so that we pass back a `Project` type rather than a `Response`, just for better type safety. Also included some logging for load errors since I was reworking some of the loading logic.

## Testing story

Tested locally.

### CODE REVIEW

File: apps/src/labs/types.ts [hunk 1]
```diff
@@ -18,6 +18,14 @@ export interface Channel {
 
 export type DefaultChannel = Pick<Channel, 'name'>;
 
+// Represents the structure of the full project sources object (i.e. the main.json file)
+export interface ProjectSources {
```
Comment (sanchitmalhotra126) [inline apps/src/labs/types.ts:22]: Couldn't think of a great name for this - basically this is the full main.json structure, whereas `Source` is just the actual sources portion (stringified for blockly/droplet labs, or a nested object for Javalab). Open to ideas :)

File: apps/src/labs/labRedux.ts [hunk 2]
```diff
@@ -233,12 +225,16 @@ function setProjectAndLevelData(
   if (aborted) {
     return;
   }
-  const {channel, source, levelData} = data;
+  const {channel, sources, levelData} = data;
   dispatch(setChannel(channel));
-  dispatch(setSource(source));
+
+  if (sources) {
```
Comment (molly-moen) [inline apps/src/labs/labRedux.ts]: if there aren't sources, should we set sources to undefined?

File: apps/src/labs/types.ts (hunk 1 above)
Comment (molly-moen) [reply apps/src/labs/types.ts:22]: seems reasonable to me

File: apps/src/music/views/MusicView.jsx [hunk 3]
```diff
@@ -401,7 +402,7 @@ class UnconnectedMusicView extends React.Component {
     }
 
     // This may no-op due to throttling.
-    this.musicBlocklyWorkspace.saveCode();
```
Comment (molly-moen) [inline apps/src/music/views/MusicView.jsx:404]: do we no longer need `MusicBlocklyWorkspace.saveCode()` (i.e. can we delete that method)?

**Review by molly-moen — COMMENTED** [review]
Looks good, a couple minor questions!

File: apps/src/music/views/MusicView.jsx (hunk 3 above)
Comment (sanchitmalhotra126) [reply apps/src/music/views/MusicView.jsx:404]: Yep! that's been deleted

File: apps/src/labs/labRedux.ts (hunk 2 above)
Comment (sanchitmalhotra126) [reply apps/src/labs/labRedux.ts]: Oh yes good call, don't want old sources to hang around.

File: apps/src/music/views/MusicView.jsx (hunk 3 above)
Comment (molly-moen) [reply apps/src/music/views/MusicView.jsx:404]: whoops don't know how I missed that

**Review by molly-moen — APPROVED** [review]

**Review by breville — APPROVED** [review]

**Pull request merged**